src/main/connect/mod.rs

```diff
@@ -202,7 +185,7 @@ impl<UID: Uid> Connect<UID> {
     fn accept(&mut self, core: &mut Core, poll: &Poll) {
         loop {
             match unwrap!(self.listener.as_ref()).accept() {
-                Ok((socket, _)) => self.exchange_msg(core, poll, Socket::wrap(socket)),
+                Ok((socket, _)) => self.exchange_msg(core, poll, TcpSock::wrap(socket)),
```

Comment:
If you are not nat-traversing here, would you even have listeners to accept something ? I don't think this function should even be around any more.

Also do check and remove `self.listener` if no longer needed.